src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePluginTest.java

```diff
@@ -1,858 +1,688 @@
-                bestCpu, bestMemory, bestDisk, imageId, null, publicKey, networksId);
-        Map<String, String> requirements = new HashMap<>();
-        requirements.put(requirementTag, requirementValue);
-        requirements.put("additionalReqTag", "additionalReqValue");
-        computeOrder.setRequirements(requirements);
-        Mockito.when(this.clientMock.doPostRequest(this.argString.capture(), this.argBodyString.capture(), this.argCloudToken.capture()
-        )).thenReturn("");
-        Mockito.when(this.launchCommandGeneratorMock.createLaunchCommand(computeOrder)).thenReturn(userData);
-        Mockito.doReturn(idKeyName).doReturn(idInstanceName).when(this.computePlugin).getRandomUUID();
-        Mockito.doReturn(expectedInstanceIdJson).when(this.clientMock).doPostRequest(argString.capture(),
-                argBodyString.capture(), argCloudToken.capture());
+    public void testFindSmallestFlavorWhenNoResourceAvailable() throws FogbowException {
+        // setup
+        ComputeOrder computeOrder = this.testUtils.createLocalComputeOrder();
+        Mockito.doReturn(null).when(this.computePlugin)
+                .getBestFlavor(Mockito.eq(computeOrder), Mockito.eq(this.cloudUser));
 
         // exercise
-        String instanceId = this.computePlugin.requestInstance(computeOrder, this.openStackV3User);
+        this.computePlugin.findSmallestFlavor(computeOrder, cloudUser);
     }
 
-    // test case: Check if a getInstance builds a compute instance from http response properly
+    // test case: test if getKeyName() returns a String
     @Test
-    public void testGetInstance() throws FogbowException, HttpResponseException {
+    public void testGetKeyName() throws FogbowException, HttpResponseException {
         // set up
-        String newComputeEndpoint = this.computeEndpoint + "/" + instanceId;
-        String computeInstanceJson = generateComputeInstanceJson(instanceId, hostName, localIpAddress, flavorId, openstackStateActive);
-        List<String> ipAddresses = new ArrayList<>();
-        ipAddresses.add(localIpAddress);
+        Mockito.doReturn(ANY_URL).when(this.computePlugin)
+                .getComputeEndpoint(Mockito.anyString(), Mockito.anyString());
 
-        ComputeOrder computeOrder = new ComputeOrder();
-        computeOrder.setInstanceId(instanceId);
+        // exercise
+        String keyName = this.computePlugin.getKeyName(this.FAKE_PROJECT_ID, cloudUser, publicKey);
 
-        ComputeInstance expectedComputeInstance = new ComputeInstance(instanceId, openstackStateActive, hostName, ipAddresses);
+        // verify
+        Mockito.verify(this.computePlugin, Mockito.times(testUtils.RUN_ONCE))
+                .getComputeEndpoint(Mockito.anyString(), Mockito.anyString());
+        Mockito.verify(this.clientMock, Mockito.times(testUtils.RUN_ONCE))
+                .doPostRequest(Mockito.anyString(), Mockito.anyString(), Mockito.eq(cloudUser));
+        Assert.assertTrue(keyName instanceof String);
```

Comment:
this is useless once you've declared it as String. There is no other type that could be converted to String by the compiler